clinica/utils/freesurfer.py

```diff
@@ -218,289 +234,129 @@ def extract_image_id_from_longitudinal_segmentation(freesurfer_id):
-                ),
-                key_list,
-                info,
-                value_list,
-            )
-
-    # Generate TSV files for segmentation files
-    #
-    # Columns in aseg.stats or wmparc.stats file
-    columns_segmentation = [
-        "Index",
-        "SegId",
-        "NVoxels",
-        "Volume_mm3",
-        "StructName",
-        "normMean",
-        "normStdDev",
-        "normMin",
-        "normMax",
-        "normRange",
-    ]
-
-    # Parsing aseg.stats
-    stats_filename = os.path.join(stats_folder, "aseg.stats")
-    df = pandas.read_csv(
-        stats_filename,
-        comment="#",
-        header=None,
-        delimiter="\s+",
-        dtype=str,
-        names=columns_segmentation,
-    )
-    secondary_stats_dict = get_secondary_stats(stats_filename, "volume")
-    key_list = list(df["StructName"]) + list(secondary_stats_dict.keys())
-    value_list = list(df["Volume_mm3"]) + list(secondary_stats_dict.values())
-    write_tsv_file(
-        os.path.join(output_dir, prefix + "_segmentationVolumes.tsv"),
-        key_list,
-        "volume",
-        value_list,
-    )
-
-    # Parsing wmparc.stats
-    stats_filename = os.path.join(stats_folder, "wmparc.stats")
-    df = pandas.read_csv(
-        stats_filename,
-        comment="#",
-        header=None,
-        delimiter="\s+",
-        dtype=str,
-        names=columns_segmentation,
-    )
-    secondary_stats_dict = get_secondary_stats(stats_filename, "volume")
-    key_list = list(df["StructName"]) + list(secondary_stats_dict.keys())
-    value_list = list(df["Volume_mm3"]) + list(secondary_stats_dict.values())
-    write_tsv_file(
-        os.path.join(output_dir, prefix + "_parcellation-wm_volume.tsv"),
-        key_list,
-        "volume",
-        value_list,
-    )
+def write_tsv_file(
+    out_filename: PurePath,
+    name_list: List[str],
+    scalar_list: List[float],
+) -> None:
+    """Write a .tsv file with list of keys and values.
 
+    Parameters
+    ----------
+    out_filename : PurePath
+        Path to the .tsv file to write to.
 
-def write_tsv_file(out_filename, name_list, scalar_name, scalar_list):
-    """Write a .tsv file with list of keys and values.
+    name_list : List[str]
+        list of keys.
 
-    Args:
-        out_filename (string): name of the .tsv file
-        name_list (list of string): list of keys
-        scalar_name (string): 'volume', 'thickness', 'area' or
-            'meanCurv'. Not used for now. Might be used as part of a
-            pandas data frame
-        scalar_list (list of float): list of values corresponding to the keys
+    scalar_list : List[float]
+        list of values corresponding to the keys.
     """
     import warnings
 
-    import pandas
-
     try:
-        data = pandas.DataFrame({"label_name": name_list, "label_value": scalar_list})
+        data = pd.DataFrame({"label_name": name_list, "label_value": scalar_list})
```

Comment:
Hmm, I agree but at the same time it seems like this is this function's only purpose to receive two lists as inputs, which kind of makes sense if you look at how it is called (on lists built by concatenating lists over a for loop with no checks on lengths).
If you change the signature, then the input that makes the most sense to me would be a dictionary. But then the whole function is useless since you can simply call `pd.DataFrame(dico)`.
I'm not even sure adding the ValueError as I did makes sense, since these kind of errors were only reported as warnings before...
Will look into it further next week...

Reply:
I deleted this function. As suspected, it wasn't needed.
Most functions in this module were doing a lot of unnecessary conversions between dictionaries, pandas dataframes, tuples of lists...
All this conversion logic should have been removed now in order to rely as much as possible on DataFrames, which hopefully makes things clearer.